### PR TITLE
HBASE-28972 Limit the number of retries in FanOutOneBlockAsyncDFSOutp…

### PR DESCRIPTION
**Pull request by Apache9**
…utHelper.completeFile

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 44s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  4s |  |  master passed  |
| +1 :green_heart: |  compile  |  12m 15s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 48s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |  12m 22s |  |  master passed  |
| +1 :green_heart: |  spotless  |   1m  8s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 27s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  13m 18s |  |  the patch passed  |
| +1 :green_heart: |  javac  |  13m 18s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   2m 15s |  |  the patch passed  |
| +1 :green_heart: |  xmllint  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  spotbugs  |  15m 57s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  16m 14s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 46s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 22s |  |  The patch does not generate ASF License warnings.  |
|  |   |  94m 42s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6453 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless xmllint |
| uname | Linux bb03787eca11 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 69ea0a486d0d69929d618b2f108e6f3bb8fbe577 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 189 (vs. ulimit of 30000) |
| modules | C: hbase-asyncfs . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/1/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 xmllint=20913 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 15s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m  6s |  |  master passed  |
| +1 :green_heart: |  compile  |   2m 13s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 24s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 25s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 54s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 13s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 13s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 18s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 32s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 246m  6s | [/patch-unit-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/1/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt) |  root in the patch failed.  |
|  |   | 278m 34s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6453 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 5802e17403b1 5.4.0-200-generic #220-Ubuntu SMP Fri Sep 27 13:19:16 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 69ea0a486d0d69929d618b2f108e6f3bb8fbe577 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/1/testReport/ |
| Max. process+thread count | 6080 (vs. ulimit of 30000) |
| modules | C: hbase-asyncfs . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 33s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 59s |  |  master passed  |
| +1 :green_heart: |  compile  |   7m 37s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m  8s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   7m 30s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 40s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 47s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   7m 36s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   7m 36s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   1m  8s |  |  the patch passed  |
| +1 :green_heart: |  xmllint  |   0m  1s |  |  No new issues.  |
| +1 :green_heart: |  spotbugs  |   7m 55s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 27s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 40s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 17s |  |  The patch does not generate ASF License warnings.  |
|  |   |  59m  2s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6453 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless xmllint |
| uname | Linux 793723f6ada6 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 69ea0a486d0d69929d618b2f108e6f3bb8fbe577 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 189 (vs. ulimit of 30000) |
| modules | C: hbase-asyncfs . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/2/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 xmllint=20913 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 46s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 18s |  |  master passed  |
| +1 :green_heart: |  compile  |   2m 17s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 20s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 30s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 55s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m  8s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m  8s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 17s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 30s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 294m 53s | [/patch-unit-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/2/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt) |  root in the patch failed.  |
|  |   | 329m 43s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/2/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6453 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux e331af28e966 5.4.0-195-generic #215-Ubuntu SMP Fri Aug 2 18:28:05 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 69ea0a486d0d69929d618b2f108e6f3bb8fbe577 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/2/testReport/ |
| Max. process+thread count | 8650 (vs. ulimit of 30000) |
| modules | C: hbase-asyncfs . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/2/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/io/asyncfs/FanOutOneBlockAsyncDFSOutputHelper.java [hunk 1]
```diff
@@ -640,8 +639,9 @@ public static boolean shouldRetryCreate(RemoteException e) {
 
   static void completeFile(FanOutOneBlockAsyncDFSOutput output, DFSClient client,
     ClientProtocol namenode, String src, String clientName, ExtendedBlock block,
-    HdfsFileStatus stat) {
-    for (int retry = 0;; retry++) {
+    HdfsFileStatus stat) throws IOException {
+    int maxRetries = client.getConf().getBlockWriteLocateFollowingInitialDelayMs();
```
Comment (stoty) [inline hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/io/asyncfs/FanOutOneBlockAsyncDFSOutputHelper.java:643]: The method name refers to milliseconds, which makes this sound we're using a temporal value for the number of retries.

Is this really the config option we want here ? Is the Hadoop getter misnamed ?

Comment (Apache9) [reply hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/io/asyncfs/FanOutOneBlockAsyncDFSOutputHelper.java:643]: Oh, should be a typo, there is a retry number config.

Let me fix.

**Review by stoty — APPROVED** [review]
+1 LGTM (pending tests)

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 42s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m  8s |  |  master passed  |
| +1 :green_heart: |  compile  |   2m  7s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 21s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 23s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  5s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m  9s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m  9s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   2m 20s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 25s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 298m  6s | [/patch-unit-root.txt](https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/3/artifact/yetus-jdk17-hadoop3-check/output/patch-unit-root.txt) |  root in the patch failed.  |
|  |   | 331m 51s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.47 ServerAPI=1.47 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/3/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6453 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 010177dbf731 5.4.0-195-generic #215-Ubuntu SMP Fri Aug 2 18:28:05 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 5875c936e646261f9ecd5d9db53e93f1ad4a6762 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/3/testReport/ |
| Max. process+thread count | 8087 (vs. ulimit of 30000) |
| modules | C: hbase-asyncfs . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6453/3/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**